lib/integrate.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2015 Mozilla
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// Import this first so we can use it to wrap other modules we import.
+var promisify = require("promisify-node");
+
+var fs = require('fs');
+var path = require('path');
+var gutil = require('gulp-util');
+
+module.exports = promisify(function(config, callback) {
+  var dir = config.dir || './';
+
+  if (!fs.existsSync(dir)) {
+    throw new Error('The directory you\'ve specified doesn\'t exist');
+  }
+
+  var offlineManagerPath = path.join(path.dirname(__dirname), 'app', 'scripts', 'offline-manager.js');
+  if (!fs.existsSync(offlineManagerPath)) {
+    throw new Error('offline-manager.js doesn\'t exist');
+  }
+
+  var dest = path.join(dir, 'offline-manager.js');
+
+  var srcStream = fs.createReadStream(offlineManagerPath);
+  srcStream.pipe(fs.createWriteStream(dest));
+  srcStream.on('end', function() {
+    gutil.log(gutil.colors.blue(
+      'Your app needs to load the script offline-manager.js in order to register the service ' +
+      'worker that offlines your app. To load the script, add this line to your app\'s HTML ' +
+      'page(s)/template(s): <script src="' + path.join(dir, 'offline-manager.js') + '"></script>'
```

Comment:
> Nit: I'd put the HTML tag on a separate line with a blank space in between, so it's easier to click-and-drag (or triple-click, on Mac) to copy it from the console window.

See for example the output of *configure*, at the end of which are these instructions, which contain embedded commands that are easier to copy/paste because they're separated from the rest of the output by blank lines:

>You're ready to auto-deploy using Travis!  Just commit the changes
>in .travis.yml and push the commit to the origin/master branch:
>
>git commit -m"configure Travis to auto-deploy to GitHub Pages" .travis.yml
>git push origin master
>
>Then visit https://travis-ci.org/mykmelez/test-app-4/builds to see the build status.

(It'd also help if the commands were a different color/style, to further set them apart, but that's #26.)